sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -39,29 +40,362 @@ class ConfigurationSetting(Model):
             last_modified=key_value.last_modified,
             tags=key_value.tags,
             read_only=key_value.locked,
             etag=key_value.etag,
         )
+
+    def _to_generated(self):
+        # type: (...) -> KeyValue
+        return KeyValue(
+            key=self.key,
+            label=self.label,
+            value=self.value,
+            content_type=self.content_type,
+            last_modified=self.last_modified,
+            tags=self.tags,
+            locked=self.read_only,
+            etag=self.etag,
+        )
+
+
+class FeatureFlagConfigurationSetting(
+    ConfigurationSetting
+):  # pylint: disable=too-many-instance-attributes
+    """A feature flag configuration value.
+    Variables are only populated by the server, and will be ignored when
+    sending a request.
+    :ivar etag: Entity tag (etag) of the object
+    :vartype etag: str
+    :ivar key:
+    :vartype key: str
+    :ivar value: The value of the configuration setting
+    :vartype value: str
+    :ivar enabled:
+    :vartype enabled: bool
+    :param filters:
+    :type filters: list[dict[str, Any]]
+    :param label:
+    :type label: str
+    :param content_type:
+    :type content_type: str
+    :ivar last_modified:
+    :vartype last_modified: datetime
+    :ivar read_only:
+    :vartype read_only: bool
+    :param tags:
+    :type tags: dict[str, str]
+    """
+
+    _attribute_map = {
+        "etag": {"key": "etag", "type": "str"},
+        "key": {"key": "key", "type": "str"},
+        "label": {"key": "label", "type": "str"},
+        "content_type": {"key": "content_type", "type": "str"},
+        "value": {"key": "value", "type": "str"},
+        "last_modified": {"key": "last_modified", "type": "iso-8601"},
+        "read_only": {"key": "read_only", "type": "bool"},
+        "tags": {"key": "tags", "type": "{str}"},
+    }
+    key_prefix = ".appconfig.featureflag/"
+    _feature_flag_content_type = (
+        "application/vnd.microsoft.appconfig.ff+json;charset=utf-8"
+    )
+    kind = "FeatureFlag"
+
+    def __init__(self, key, enabled, filters=None, **kwargs):
+        # type: (str, bool, Optional[List[Dict[str, Any]]]) -> None
+        self._enabled = enabled
+        super(FeatureFlagConfigurationSetting, self).__init__(**kwargs)
+        self.key = key
+        self.label = kwargs.get("label", None)
+        self.content_type = kwargs.get("content_type", self._feature_flag_content_type)
+        self.last_modified = kwargs.get("last_modified", None)
+        self.tags = kwargs.get("tags", {})
+        self.read_only = kwargs.get("read_only", None)
+        self.etag = kwargs.get("etag", None)
+        self.description = kwargs.get("description", None)
+        self.display_name = kwargs.get("display_name", None)
+        self._filters = filters or []
+        self._value = kwargs.get("value", {})
+
+    @property
+    def enabled(self):
+        # type: () -> bool
+        if self._value is None:
```

Comment:
Not  raise AttributeError("value parameter is expected to be a dictionary") if value is None?

Reply:
Are you suggesting that we raise that error if the user tries to set the value to None?

Reply:
If _value is not in right schema or key not startswith(self.key_prefix), it is not FeatureFlagConfigurationSetting any longer. Hence enabled & filters properties are invalid.

Reply:
self._value is the real value we store. 

In a GETTER, we cannot change the value.

```py
if self._value is None:
    return None
```

same in 
```py
else:
            self._value["enabled"] = self._enabled
```

the value can only be changed in a setter method which means user explicitly modify the value.

Reply:
And I would suggest we have a private method to validate if the value and key are valid. It will be called for multiple times. (any time user reads FeatureFlagConfigurationSetting specific properties, we need to validate it first.)

Reply:
To be clear, we will
1. validate anytime `my_key.value = <new_value>` and `my_key.enabled = <bool>` occur, (validating on a set)?
2. Validate anytime a user tries to read the `value` or `enabled` properties as well (validate on a get)?
3. If they are valid, we update the internal dictionary or enabled value 
4. if they are not valid (ie not a dictionary, not a boolean) we raise an error?